src/View/WorkController.ts

```diff
@@ -14,51 +14,61 @@ export interface Arguments {
   readonly dependencies: Dependencies
 }
 
-export class WorkController {
-  private readonly logger: Pino.Logger
-  private readonly db: Db
-  private readonly anchorCollection: Collection
-  private readonly workCollection: Collection
-
-  constructor({
-    dependencies: {
-      logger,
-      db,
-    },
-  }: Arguments) {
-    this.logger = childWithFileName(logger, __filename)
-    this.db = db
-    this.anchorCollection = this.db.collection('anchors')
-    this.workCollection = this.db.collection('works')
-  }
+export interface WorkController {
+  readonly createWork: (work: Work) => Promise<void>
+  readonly setIPFSHash: (workId: string, ipfsFileHash: string) => Promise<void>
+  readonly setDirectoryHashOnEntries: (data: {
+    ipfsFileHashes: ReadonlyArray<string>
+    ipfsDirectoryHash: string,
+  }) => Promise<void>
+  readonly setTxId: (ipfsDirectoryHash: string, transactionId: string) => Promise<void>
+  readonly upsertAnchors: (poetAnchors: ReadonlyArray<PoetBlockAnchor>) => Promise<void>
+  readonly setFileHashesForDirectoryHash: (data: {
+    ipfsFileHashes: ReadonlyArray<string>
+    ipfsDirectoryHash: string,
+  }) => Promise<void>
+  readonly upsertClaimIPFSHashPair: (claimIPFSHashPairs: ReadonlyArray<ClaimIPFSHashPair>) => Promise<void>
+}
+
+export const WorkController = ({
+  dependencies: {
+    logger,
+    db,
+  },
+}: Arguments): WorkController => {
+  const workControllerLogger: Pino.Logger = childWithFileName(logger, __filename)
```

Comment:
I'm not a big fan of shadowing function arguments in general but I think in this case it could make sense... 

1. is there any place other than this line of code in which we'd want to access the passed logger rather than the `workControllerLogger`?
2. aren't we shadowing it anyway, inside the scope of the functions declared here?